### ISSUE TITLE
use async.io for asynchronous path finding

### ISSUE THREAD

**Issue by brean** [brean/python-pathfinding#5]
inspired by [easystarjs](https://github.com/prettymuchbryce/easystarjs)

**Comment by brean** [comment]
Also python 2 will die [soon](https://pythonclock.org/) changing to the async-await syntax should not be a problem!

**Comment by ClementJ18** [comment]
From my understanding, asyncio in python is useful when you have IO bound operations such as reading files or sending HTTP requests. Pathfinding doesn't really have any of this as far as I can tell so don't think async would be a good match for it.

Threading seems like it would potentially be a better match although that's a whole different can of worms.

**Comment by brean** [comment]
You are right, multiprocessing would make more sense for cpu-bound operations like this.
